Tests/DiligentCoreTest/src/ShaderTools/ShaderPreprocessTest.cpp

```diff
@@ -123,13 +123,43 @@ TEST(ShaderPreprocessTest, Include)
         ShaderCI.FilePath                   = "IncludeCommentsTrickyCasesTest.hlsl";
         ShaderCI.pShaderSourceStreamFactory = pShaderSourceFactory;
 
-        const auto Result = ProcessShaderIncludes(ShaderCI, [&](const ProcessShaderIncludesInfo& Info) {
-            EXPECT_EQ(SafeStrEqual(Info.FilePath, Includes.front()), true);
-            Includes.pop_front();
-        });
+        const auto Result = ProcessShaderIncludes(ShaderCI, {[&](IDataBlob* pDataBlob, const char* FilePath) {
+                                                      EXPECT_EQ(SafeStrEqual(FilePath, Includes.front()), true);
+                                                      Includes.pop_front();
+                                                  }});
 
         EXPECT_EQ(Result, true);
     }
 }
 
+TEST(ShaderPreprocessTest, MergeIncludes)
+{
+    RefCntAutoPtr<IShaderSourceInputStreamFactory> pShaderSourceFactory;
+    CreateDefaultShaderSourceStreamFactory("shaders/ShaderPreprocessor", &pShaderSourceFactory);
+    ASSERT_NE(pShaderSourceFactory, nullptr);
+
+    {
+        ShaderCreateInfo ShaderCI{};
+        ShaderCI.Desc.Name                  = "TestShader";
+        ShaderCI.FilePath                   = "InlineIncludeShaderTest.hlsl";
+        ShaderCI.pShaderSourceStreamFactory = pShaderSourceFactory;
+
+        const std::string Reference =
+            "// InlineIncludeShaderCommon0.hlsl\n\n"
```

Comment:
Can you add include in the comment to the test, e.g.
/* #include "Non-existing" */
Also, please add `#include <>`